pkg/apis/pxc/v1/pxc_types.go

```diff
@@ -439,9 +439,9 @@ func (cr *PerconaXtraDBCluster) setSecurityContext() {
 		cr.Spec.ProxySQL.PodSecurityContext = sc
 	}
 	if cr.Spec.Backup != nil {
-		for k := range cr.Spec.Backup.Storages {
-			if cr.Spec.Backup.Storages[k].PodSecurityContext == nil {
-				cr.Spec.Backup.Storages[k].PodSecurityContext = sc
+		for _, storage := range cr.Spec.Backup.Storages {
+			if storage.PodSecurityContext == nil {
+				storage.PodSecurityContext = sc
```

Comment:
Revert. please
You change a copy here

Reply:
You are right, this will not work. But a revert is not enough because `Storage` have to be a map with a non-pointer value. Otherwise osdk reports `map values must be a named type, not *ast.StarExpr`

How about this workaround?

Reply:
> But a revert is not enough because Storage have to be a map with a non-pointer value.
It was a pointer value unlit you have changed it.